src/interpreter/types/interpreter.ts

```diff
@@ -0,0 +1,40 @@
+import { type Stack } from '../../utils/stack';
+import { type Value } from './evaluationResults';
+import {
+  type BlockOrEmptyStatement,
+  type Identifier,
+  type Node
+} from '../../ast/types';
+import { type VirtualMemory } from '../../memory/virtualMemory';
+import { type Instr } from './instruction';
+import { type TextMemory } from '../../memory/textMemory';
+
+export interface ExplicitControlEvaluatorState {
+  agenda: Stack<AgendaItem>;
+  stash: Stack<Value>;
+  environment: Environment;
+  memory: VirtualMemory;
+  textMemory: TextMemory;
```

Comment:
I guess this is fine for now, but eventually it would be nice if we had a singular memory object that we can access, even if it's just a facade over what we currently have.

Reply:
I changed this to a singular memory object.